lib/index.js

```diff
@@ -111,130 +108,141 @@ function getPolyfillString(options) {
-								if (depName in allFeatures) {
-									graph.add(depName, featureName);
-								}
+			// Outer closure hides private features from global scope
+			output.queue("(function(undefined) {" + lf);
+
+			if (options.rum && process.env.RUM_BEACON_HOST) {
+				output.queue(() => {
+					const optionsForRUM = cloneDeep(options);
+					Object.keys(options.features).forEach(f => optionsForRUM.features[f].flags.add('always'));
+					return getPolyfills(optionsForRUM)
+						.then(rumFeatures => {
+							return rumTemplate({
+								host: process.env.RUM_BEACON_HOST,
+								features: Object.keys(rumFeatures).reduce((op, featureName) => {
+									const polyfill = sourceslib.getPolyfillMetaSync(featureName);
+									if (polyfill.detectSource) {
+										op.push({name: featureName, detect: polyfill.detectSource});
+									}
+									return op;
+								}, [])
 							});
-						}
-
-						feature.comment = '- ' + featureName + ', License: ' + (polyfill.license || 'CC0') + ((feature.aliasOf && feature.aliasOf.length) ? ' (required by "' + feature.aliasOf.join('", "') + '")' : '');
-					}
+						})
+					;
 				});
-			}))
-			.then(() => {
-
-				// Using the graph order, build the completed polyfill bundle
-				graph.sort().forEach(featureName => {
-					if (targetedFeatures.includes(featureName)) {
-						builtPolyfillString += allFeatures[featureName].polyfillOutput || '';
-						if (!options.minify && allFeatures[featureName].comment) {
-							explainerComment.push(allFeatures[featureName].comment);
-						}
-					}
-				});
-
-				if (builtPolyfillString) {
-					let rum = '';
-					if (options.rum && process.env.RUM_BEACON_HOST) {
-						rum = rumTemplate({host: process.env.RUM_BEACON_HOST, features:allFeatures});
-					}
-
-					// Outer closure hides private features from global scope
-					builtPolyfillString = "(function(undefined) {" + lf + rum + builtPolyfillString + lf + "})" + lf;
-
-					// Invoke the closure, binding `this` to window (in a browser), self (in a web worker), or global (in Node/IOjs)
-					builtPolyfillString += ".call('object' === typeof window && window || 'object' === typeof self && self || 'object' === typeof global && global || {});";
-				}
+			}
 
-				if (warnings.unknown.length && !options.minify) {
-					explainerComment = explainerComment.concat(
-						'',
-						'These features were not recognised:',
-						warnings.unknown
-					);
-				}
+			// Using the graph, stream all the polyfill sources in dependency order
+			sortedFeatures.forEach(featureName => {
+				const polyfill = sourceslib.getPolyfillMetaSync(featureName);
+				const wrapInDetect = (targetedFeatures[featureName].flags.has('gated') && polyfill.detectSource);
+				
+				if (wrapInDetect) output.queue("if (!(" + polyfill.detectSource + ")) {" + lf);
+				output.queue(() => {
+					return sourceslib.streamPolyfillSource(featureName, (options.minify ? 'min':'raw'))
+				});
+				if (wrapInDetect) output.queue(lf + "}" + lf + lf);
```

Comment:
This is quite a nice example of weaving together strings and a lazy-loaded stream.